Integrations/PaloAltoNetworksCortex/PaloAltoNetworksCortex.py

```diff
@@ -608,32 +625,52 @@ def process_incident_pairs(incident_pairs, max_incidents):
 
 
 def fetch_incidents():
-    last_fetched_event_timestamp = demisto.getLastRun().get('last_fetched_event_timestamp')
-    if last_fetched_event_timestamp is not None:
-        last_fetched_event_timestamp = datetime.strptime(last_fetched_event_timestamp, '%Y-%m-%dT%H:%M:%S.%f')
+
+    last_run = demisto.getLastRun()
+    last_fetched_event_timestamp = last_run.get('last_fetched_event_timestamp')
+    last_query_id = last_run.get('last_query_id')
+
+    if last_query_id:
+        # Need to poll query results fron last run
+        response = poll_query_result(last_query_id)
     else:
-        last_fetched_event_timestamp, _ = parse_date_range(FIRST_FETCH_TIMESTAMP)
+        if last_fetched_event_timestamp is not None:
+            last_fetched_event_timestamp = datetime.strptime(last_fetched_event_timestamp, '%Y-%m-%dT%H:%M:%S.%f')
+        else:
+            last_fetched_event_timestamp, _ = parse_date_range(FIRST_FETCH_TIMESTAMP)
 
-    # Need sometime in the future, so the timestamp will be taken from the query
-    service_end_date_epoch = int(datetime.now().strftime('%s')) + 1000
+        # Need sometime in the future, so the timestamp will be taken from the query
+        service_end_date_epoch = int(datetime.now().strftime('%s')) + 1000
 
-    if 'Firewall' in FETCH_QUERY:  # type: ignore
-        fetch_timestamp = int(last_fetched_event_timestamp.strftime('%s'))
-    elif 'Traps' in FETCH_QUERY:  # type: ignore
-        fetch_timestamp = last_fetched_event_timestamp.strftime('%Y-%m-%dT%H:%M:%S.%fZ')
+        if 'Firewall' in FETCH_QUERY:  # type: ignore
+            fetch_timestamp = int(last_fetched_event_timestamp.strftime('%s'))
+        elif 'Traps' in FETCH_QUERY:  # type: ignore
+            fetch_timestamp = last_fetched_event_timestamp.strftime('%Y-%m-%dT%H:%M:%S.%fZ')
 
-    query = prepare_fetch_query(fetch_timestamp)
+        query = prepare_fetch_query(fetch_timestamp)
 
-    query_data = {
-        'query': query,
-        'startTime': 0,
-        'endTime': service_end_date_epoch,
-    }
+        query_data = {
+            'query': query,
+            'startTime': 0,
+            'endTime': service_end_date_epoch,
+        }
 
-    response = query_loggings(query_data)
+        response = query_loggings(query_data)
 
     try:
-        result = response.json()['result']
+        response_json = response.json()
+        query_status = response_json.get('queryStatus', '')
+        if query_status == 'JOB_FAILED':
+            raise Exception('Logging query job failed with status: JOB_FAILED')
+        elif query_status == 'RUNNING':
+            # If query job is still running after 30 seconds (max timeout), pass it to next run
+            demisto.setLastRun({
+                'last_fetched_event_timestamp': last_fetched_event_timestamp,
```

Comment:
Shouldn't you be doing: strftime('%Y-%m-%dT%H:%M:%S.%f') as is done in the non `RUNNING` case?